# Analiza Danych dotyczących rynku aplikacji mobilnych
Projket ten ma za zadanie pokazać jak powinna wyglądać analiza danych dla filmy zajmującej się tworzeniem darmowych aplikacji, której głównym źródłem finansowania będą reklamy. Oznacza to że ilość użytkowników aplikacji determinuje zarobki.

Celem projektu jest zrozumienie jaki typ aplikacji jest chętniej wybierany przez użytkowników.

# Zbiór Danych

Według danych z 2018 roku, w Google Play było 2.1 miliona a w AppStore 2 miliony aplikacji mobilnych:

<center>

![Image](https://s3.amazonaws.com/dq-content/350/py1m8_statista.png)

Źródło: [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)

</center>

Z uwagi na czas, koszty oraz dostępność, poslużono się mniejszymi zbiorami danych o aplikacjach mobilnych:


<center>

[Dane Aplikacji Google Play](https://www.kaggle.com/datasets/lava18/google-play-store-apps)

[Dane Aplikacji App Store](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps)

</center>



## Pobranie danych:

In [ ]:
!wget https://dq-content.s3.amazonaws.com/350/googleplaystore.csv -O google_wget.csv
!wget https://dq-content.s3.amazonaws.com/350/AppleStore.csv -O apple_wget.csv
from csv import reader
android = list(reader(open('google_wget.csv')))
android_header = android[0]
android = android[1:]

ios = list(reader(open('apple_wget.csv')))
ios_header = ios[0]
ios = ios[1:]


--2022-11-14 09:22:06--  https://dq-content.s3.amazonaws.com/350/googleplaystore.csv
Resolving dq-content.s3.amazonaws.com (dq-content.s3.amazonaws.com)... 52.216.26.76
Connecting to dq-content.s3.amazonaws.com (dq-content.s3.amazonaws.com)|52.216.26.76|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1349314 (1.3M) [text/csv]
Saving to: ‘google_wget.csv’

google_wget.csv     100%[===================>]   1.29M  --.-KB/s    in 0.1s    

2022-11-14 09:22:06 (11.3 MB/s) - ‘google_wget.csv’ saved [1349314/1349314]

--2022-11-14 09:22:06--  https://dq-content.s3.amazonaws.com/350/AppleStore.csv
Resolving dq-content.s3.amazonaws.com (dq-content.s3.amazonaws.com)... 52.216.26.76
Connecting to dq-content.s3.amazonaws.com (dq-content.s3.amazonaws.com)|52.216.26.76|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 725761 (709K) [text/csv]
Saving to: ‘apple_wget.csv’

apple_wget.csv      100%[===================>] 708.75K  --.-KB/s    in 0.1s    



In [ ]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [ ]:
def explore_data_pandas(dane1, start, end):
  print(f'Liczba wierszy: {len(dane1)}')
  print(f'Liczba kolumn: {len(dane1[0])}')
  import pandas
  df = pandas.DataFrame(dane1[start:end])
  display(df)

In [ ]:
explore_data_pandas(android, 0, 12)

Liczba wierszy: 10841
Liczba kolumn: 13


0               1    2   \
0      Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN  4.1   
1                                 Coloring book moana  ART_AND_DESIGN  3.9   
2   U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN  4.7   
3                               Sketch - Draw & Paint  ART_AND_DESIGN  4.5   
4               Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN  4.3   
5                          Paper flowers instructions  ART_AND_DESIGN  4.4   
6             Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN  3.8   
7                                    Infinite Painter  ART_AND_DESIGN  4.1   
8                                Garden Coloring Book  ART_AND_DESIGN  4.4   
9                       Kids Paint Free - Drawing Fun  ART_AND_DESIGN  4.7   
10                            Text on Photo - Fonteee  ART_AND_DESIGN  4.4   
11            Name Art Photo Editor - Focus n Filters  ART_AND_DESIGN  4.4   

        3     4            5     6  7         8                          9   \
0      159   19M      10,000+  Free  0  Everyone               Art & Design   
1      967   14M     500,000+  Free  0  Everyone  Art & Design;Pretend Play   
2    87510  8.7M   5,000,000+  Free  0  Everyone               Art & Design   
3   215644   25M  50,000,000+  Free  0      Teen               Art & Design   
4      967  2.8M     100,000+  Free  0  Everyone    Art & Design;Creativity   
5      167  5.6M      50,000+  Free  0  Everyone               Art & Design   
6      178   19M      50,000+  Free  0  Everyone               Art & Design   
7    36815   29M   1,000,000+  Free  0  Everyone               Art & Design   
8    13791   33M   1,000,000+  Free  0  Everyone               Art & Design   
9      121  3.1M      10,000+  Free  0  Everyone    Art & Design;Creativity   
10   13880   28M   1,000,000+  Free  0  Everyone               Art & Design   
11    8788   12M   1,000,000+  Free  0  Everyone               Art & Design   

                    10                  11            12  
0      January 7, 2018               1.0.0  4.0.3 and up  
1     January 15, 2018               2.0.0  4.0.3 and up  
2       August 1, 2018               1.2.4  4.0.3 and up  
3         June 8, 2018  Varies with device    4.2 and up  
4        June 20, 2018                 1.1    4.4 and up  
5       March 26, 2017                 1.0    2.3 and up  
6       April 26, 2018                 1.1  4.0.3 and up  
7        June 14, 2018            6.1.61.1    4.2 and up  
8   September 20, 2017               2.9.2    3.0 and up  
9         July 3, 2018                 2.8  4.0.3 and up  
10    October 27, 2017               1.0.4    4.1 and up  
11       July 31, 2018              1.0.15    4.0 and up

In [ ]:
explore_data_pandas(ios, 0, 10)

Liczba wierszy: 7197
Liczba kolumn: 16


0                        1          2    3    4        5     6   \
0  284882215                 Facebook  389879808  USD  0.0  2974676   212   
1  389801252                Instagram  113954816  USD  0.0  2161558  1289   
2  529479190           Clash of Clans  116476928  USD  0.0  2130805   579   
3  420009108               Temple Run   65921024  USD  0.0  1724546  3842   
4  284035177  Pandora - Music & Radio  130242560  USD  0.0  1126879  3594   
5  429047995                Pinterest   74778624  USD  0.0  1061624  1814   
6  282935706                    Bible   92774400  USD  0.0   985920  5320   
7  553834731         Candy Crush Saga  222846976  USD  0.0   961794  2453   
8  324684580            Spotify Music  132510720  USD  0.0   878563  8253   
9  343200656              Angry Birds  175966208  USD  0.0   824451   107   

    7    8        9    10                 11  12 13  14 15  
0  3.5  3.5     95.0   4+  Social Networking  37  1  29  1  
1  4.5  4.0    10.23  12+      Photo & Video  37  0  29  1  
2  4.5  4.5  9.24.12   9+              Games  38  5  18  1  
3  4.5  4.0    1.6.2   9+              Games  40  5   1  1  
4  4.0  4.5    8.4.1  12+              Music  37  4   1  1  
5  4.5  4.0     6.26  12+  Social Networking  37  5  27  1  
6  4.5  5.0    7.5.1   4+          Reference  37  5  45  1  
7  4.5  4.5  1.101.0   4+              Games  43  5  24  1  
8  4.5  4.5    8.4.3  12+              Music  37  5  18  1  
9  4.5  3.0    7.4.0   4+              Games  38  0  10  1

## Usuwanie błędnego wiersza

Wiersz 10472 jest błędny przez brak kategorii, należy go usunąć ze zbioru danych

In [ ]:
print(android[10472])  # wiersz błędny
print('\n')
print(android_header)  # header

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


## Testy metody
Zostawiam je w celach dydaktycznych

In [ ]:
lista = [1,2,3,4,5]
lista3 = [x if x%2 else 0 for x in lista]
print(lista3)

[1, 0, 3, 0, 5]


In [ ]:
LL = [[1,2], [3,4],[5,6]]
print(LL)
# Chcemy uzyskac LL_N = [[1,2],[5,6]]
LL_2 = [ [b for b in a] for a in LL]
print(LL_2)


LL_3 = [[b != 3 for b in a] for a in LL]
print(LL_3)


LL_4 = [[b for b in a if b != 3] for a in LL]
print(LL_4)

LL_5 = [a for a in  LL if a[0] != 3 ]
print(LL_5)

[[1, 2], [3, 4], [5, 6]]
[[1, 2], [3, 4], [5, 6]]
[[True, True], [False, True], [True, True]]
[[1, 2], [4], [5, 6]]
[[1, 2], [5, 6]]


In [ ]:
new_android =[a for a in  android if a[0] != 'Life Made WI-Fi Touchscreen Photo Frame' ]
print(new_android[10471])
print(new_android[10472])
print(new_android[10473])

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']
['Sat-Fi Voice', 'COMMUNICATION', '3.4', '37', '14M', '1,000+', 'Free', '0', 'Everyone', 'Communication', 'November 21, 2014', '2.2.1.5', '2.2 and up']


### Sposób 2

In [ ]:
from numpy.lib.function_base import append
new_android_2 = []
for app in android:
    name = app[0]
    if name != 'Life Made WI-Fi Touchscreen Photo Frame':
        new_android_2.append(app)
print(new_android_2[10471])
print(new_android_2[10472])
print(new_android_2[10473])

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']
['Sat-Fi Voice', 'COMMUNICATION', '3.4', '37', '14M', '1,000+', 'Free', '0', 'Everyone', 'Communication', 'November 21, 2014', '2.2.1.5', '2.2 and up']


#### Usuwanie powtórzeń




### 1

Iteracja po całym zbiorze pokazuje ze np aplikacja Instagram pojawia sie 4 razy. Wiersze różni ilość ocen, wyciąganie wiersza o największej liczbie ocen będzie opisane niżej

In [ ]:
for app in new_android:
    nazwa = app[0]
    if nazwa == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [ ]:
android_duplikacje = []
android_bez_duplikacji = []

for app in new_android:
    nazwa = app[0]
    if nazwa in android_bez_duplikacji:
        android_duplikacje.append(nazwa)
    else:
        android_bez_duplikacji.append(nazwa)

print('Ilosc duplikacji:', len(android_duplikacje))
print('\n')
print('Przykladowe duplikacje:', android_duplikacje[:15])

Ilosc duplikacji: 1181


Przykladowe duplikacje: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']



Należy pozostawić wiersze o najwięskzej liczbie ocen.

Cele:
- Stworzyć słownik w którym każdy klucz będzie miał swoją nazwę aplikacji i przypisana do niego wartość będzie największą liczbą ocen ze zbioru.
- Użyć słownika w celu utworzenia zbioru danych bez duplikacji i z maksymalną liczbą ocen.

### 2

Budowa słownika:

In [ ]:
oceny_max = {}

for app in new_android:
    nazwa=app[0]
    liczba_ocen = float(app[3])
    if nazwa in oceny_max and oceny_max[nazwa] < liczba_ocen:
        oceny_max[nazwa] = liczba_ocen
    elif nazwa not in oceny_max:
        oceny_max[nazwa] = liczba_ocen

Zobaczmy jak wyglądają wartości w naszym słowniku

In [ ]:
xx = list(oceny_max.items())[0:3]
print(*xx, sep="\n")

('Photo Editor & Candy Camera & Grid & ScrapBook', 159.0)
('Coloring book moana', 974.0)
('U Launcher Lite – FREE Live Cool Themes, Hide Apps', 87510.0)


Sprawdzenie czy nowa lista aplikacji bez duplikacji sie zgadza

In [ ]:
print('Ile być powinno:', len(new_android) - 1181)
print('Ile jest:', len(oceny_max))

Ile być powinno: 9659
Ile jest: 9659


Słownik posłuży nam do stworzenia zbioru danych bez duplikacji:

Dla każdego wiersza w naszym zbiorze danych, porównujemy czy liczba ocen zgdza się z liczbą ocen z naszego słownika ocen maksymalnych. Dodatkowo musimy sprawdzić czy dany wiersz nie został już wcześniej dodany. W tym celu powstała 2 lista juz_dodane.

In [ ]:
android_czysty = []
juz_dodane = []

for app in new_android:
    nazwa = app[0]
    liczba_ocen = float(app[3])

    if (oceny_max[nazwa] == liczba_ocen) and (nazwa not in juz_dodane):
        android_czysty.append(app)
        juz_dodane.append(nazwa)

In [ ]:
explore_data_pandas(android_czysty, 1, 10)

Liczba wierszy: 9659
Liczba kolumn: 13


0               1    2   \
0  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN  4.7   
1                              Sketch - Draw & Paint  ART_AND_DESIGN  4.5   
2              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN  4.3   
3                         Paper flowers instructions  ART_AND_DESIGN  4.4   
4            Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN  3.8   
5                                   Infinite Painter  ART_AND_DESIGN  4.1   
6                               Garden Coloring Book  ART_AND_DESIGN  4.4   
7                      Kids Paint Free - Drawing Fun  ART_AND_DESIGN  4.7   
8                            Text on Photo - Fonteee  ART_AND_DESIGN  4.4   

       3     4            5     6  7         8                        9   \
0   87510  8.7M   5,000,000+  Free  0  Everyone             Art & Design   
1  215644   25M  50,000,000+  Free  0      Teen             Art & Design   
2     967  2.8M     100,000+  Free  0  Everyone  Art & Design;Creativity   
3     167  5.6M      50,000+  Free  0  Everyone             Art & Design   
4     178   19M      50,000+  Free  0  Everyone             Art & Design   
5   36815   29M   1,000,000+  Free  0  Everyone             Art & Design   
6   13791   33M   1,000,000+  Free  0  Everyone             Art & Design   
7     121  3.1M      10,000+  Free  0  Everyone  Art & Design;Creativity   
8   13880   28M   1,000,000+  Free  0  Everyone             Art & Design   

                   10                  11            12  
0      August 1, 2018               1.2.4  4.0.3 and up  
1        June 8, 2018  Varies with device    4.2 and up  
2       June 20, 2018                 1.1    4.4 and up  
3      March 26, 2017                 1.0    2.3 and up  
4      April 26, 2018                 1.1  4.0.3 and up  
5       June 14, 2018            6.1.61.1    4.2 and up  
6  September 20, 2017               2.9.2    3.0 and up  
7        July 3, 2018                 2.8  4.0.3 and up  
8    October 27, 2017               1.0.4    4.1 and up

## Usuwanie nie anglojęzycznych aplikacji
Każdy znak ma swój numer, sprawdza się go funkcją `ord()`
Korzystając z wiedzy że anglojęzyczny alfabet ma znaki w zakresie od 0 do 127.
W tym celu stworzona jest funkcja `czy_po_angielsku()`

In [ ]:
def czy_po_angielsku(string):

    for znak in string:
        if ord(znak) > 127:
            return False

    return True

print(czy_po_angielsku('Instagram'))
print(czy_po_angielsku('爱奇艺PPS -《欢乐颂2》电视剧热播'))


True
False


Należy pamiętać że niektóre aplikacje posaidają emotikony lub inne znaki spoza zakresu 0-127. Uznano, że aplikacja anglojęzyczna może posiadać 3 takei znaki.

Poniżej znajduje się ulepszona funkcja `czy_po_angielsku()`

In [ ]:
def czy_po_angielsku(string):
    poza_zakresem = 0

    for znak in string:
        if ord(znak) > 127:
            poza_zakresem += 1

    if poza_zakresem > 3:
        return False
    else:
        return True

print(czy_po_angielsku('Docs To Go™ Free Office Suite'))
print(czy_po_angielsku('Instachat 😜'))

True
True


Czas na zastosowanie funkcji na naszych zbiorach danych i stworzenie nowych, anglojęzycznych.

In [ ]:
android_po_angielsku = []
ios_po_angielsku = []

for app in android_czysty:
    nazwa = app[0]
    if czy_po_angielsku(nazwa):
        android_po_angielsku.append(app)

for app in ios:
    nazwa = app[1]
    if czy_po_angielsku(nazwa):
        ios_po_angielsku.append(app)

explore_data_pandas(android_po_angielsku, 0, 6)
print('\n')
explore_data_pandas(ios_po_angielsku, 0, 6)

Liczba wierszy: 9614
Liczba kolumn: 13


0               1    2   \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN  4.1   
1  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN  4.7   
2                              Sketch - Draw & Paint  ART_AND_DESIGN  4.5   
3              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN  4.3   
4                         Paper flowers instructions  ART_AND_DESIGN  4.4   
5            Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN  3.8   

       3     4            5     6  7         8                        9   \
0     159   19M      10,000+  Free  0  Everyone             Art & Design   
1   87510  8.7M   5,000,000+  Free  0  Everyone             Art & Design   
2  215644   25M  50,000,000+  Free  0      Teen             Art & Design   
3     967  2.8M     100,000+  Free  0  Everyone  Art & Design;Creativity   
4     167  5.6M      50,000+  Free  0  Everyone             Art & Design   
5     178   19M      50,000+  Free  0  Everyone             Art & Design   

                10                  11            12  
0  January 7, 2018               1.0.0  4.0.3 and up  
1   August 1, 2018               1.2.4  4.0.3 and up  
2     June 8, 2018  Varies with device    4.2 and up  
3    June 20, 2018                 1.1    4.4 and up  
4   March 26, 2017                 1.0    2.3 and up  
5   April 26, 2018                 1.1  4.0.3 and up



Liczba wierszy: 6183
Liczba kolumn: 16


0                        1          2    3    4        5     6   \
0  284882215                 Facebook  389879808  USD  0.0  2974676   212   
1  389801252                Instagram  113954816  USD  0.0  2161558  1289   
2  529479190           Clash of Clans  116476928  USD  0.0  2130805   579   
3  420009108               Temple Run   65921024  USD  0.0  1724546  3842   
4  284035177  Pandora - Music & Radio  130242560  USD  0.0  1126879  3594   
5  429047995                Pinterest   74778624  USD  0.0  1061624  1814   

    7    8        9    10                 11  12 13  14 15  
0  3.5  3.5     95.0   4+  Social Networking  37  1  29  1  
1  4.5  4.0    10.23  12+      Photo & Video  37  0  29  1  
2  4.5  4.5  9.24.12   9+              Games  38  5  18  1  
3  4.5  4.0    1.6.2   9+              Games  40  5   1  1  
4  4.0  4.5    8.4.1  12+              Music  37  4   1  1  
5  4.5  4.0     6.26  12+  Social Networking  37  5  27  1

Na ten moment posiadamy anglojęzyczne zbiory bez powtórzeń:

Android: 9614 aplikacji

IOS: 6183 aplikacji

## Usuwanie płatnych aplikacji



In [ ]:
android_darmowe = []
ios_darmowe = []

for app in android_po_angielsku:
    cena = app[7]
    if cena == '0':
        android_darmowe.append(app)

for app in ios_po_angielsku:
    cena = app[4]
    if cena == '0.0':
        ios_darmowe.append(app)

print('Ilosc aplikacji na androida ' + str(len(android_darmowe)))
print(f'Ilosc aplikacji na IOS {len(ios_darmowe)}')

Ilosc aplikacji na androida 8864
Ilosc aplikacji na IOS 3222


## Najpolarniejsze Aplikacje ze względu na rodzaj.
Zajmijmy się stworzeniem funkcji które pomogą nam zobrazować częstość występowania danego typu aplikacji.

- pierwsza funkcja pokaże procentowy udział
- druga ułoży dane malejąco

In [ ]:
def tabela_frekwencji(dataset, index):
    tabela = {}
    w_sumie = 0

    for wiersz in dataset:
        w_sumie += 1 #aby potem podzielic przez calosc
        wartosc = wiersz[index]
        if wartosc in tabela:
            tabela[wartosc] += 1 #jezeli jest to dodaj 1 do wartosci
        else:
            tabela[wartosc] = 1 #jezeli nie ma to wartosc = 1

    tabela_procentowa = {}
    for klucz in tabela:
        procent = (tabela[klucz] / w_sumie) * 100 #ilosc dodanych przez wszytskie
        tabela_procentowa[klucz] = procent #

    return tabela_procentowa


def wyswietl_tabele(dataset, index):
    tabela = tabela_frekwencji(dataset, index)
    tw = [] #pusta lista
    for klucz in tabela: #klucz to nazwa aplikacji
        klucz_wartosc = (tabela[klucz], klucz) #
        tw.append(klucz_wartosc)

    posortowana_tabela = sorted(tw, reverse = True)
    for wartosc in posortowana_tabela:
        print(wartosc[1], ':', wartosc[0])


In [ ]:
wyswietl_tabele(ios_darmowe, -5) #kolumna prime_genre IOS

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In [ ]:
wyswietl_tabele(android_darmowe, 1) #kolumna category Android

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [ ]:
wyswietl_tabele(android_darmowe, -4) #kolumna genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Populaność aplikacji czyli to ile ma ona uzytkownikow sprawdzic mozna liczbą instalacji, w zbiorze danych AppStore nie ma takiej informacji, użyta zostanie kolumna `rating_count_tot` w któ®ej znajdują się infoamcje o całkowitej liczbie ocen danej aplikacji.

### App Store


In [ ]:
oceny_total_ios = tabela_frekwencji(ios_darmowe, -5) # -5 to prime_genre
#print('>>>')
#print(oceny_total_ios)
for rodzaj in oceny_total_ios:
  #
    razem = 0
    dlugosc_r = 0
    for app in ios_darmowe:
        rodzaj_apki = app[-5]
        if rodzaj_apki == rodzaj:
            n_ocen = float(app[5]) # 5 to ilosc ocen
            razem += n_ocen #za kazdym przejsciem pętli gdy znajdzie się
            dlugosc_r += 1
    #print('razem:')
    #print(razem)
    #print('dlugosc r:')
    #rint(dlugosc_r)
    sr_n_ocen = razem / dlugosc_r
    print(rodzaj, ':', sr_n_ocen)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


### Android
W zbiorze danych istanieje kolumna `Installs`

In [ ]:
wyswietl_tabele(android_darmowe, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


W danych jest dość spora dysproporcja, rangi rosną wykładniczo i ciężko o miarodajność, ponizej podwójną funckją `for` obliczono średnią ilość instalacji dla danej kategorii aplikacji.

In [ ]:
kategorie_android = tabela_frekwencji(android_darmowe, 1)

for kategoria in kategorie_android:
    razem = 0
    dl_zbioru_kategorii = 0
    for app in android_darmowe:
        categoria_apki = app[1]
        if categoria_apki == kategoria:
            n_istalacji = app[5]
            n_istalacji = n_istalacji.replace(',', '')
            n_istalacji = n_istalacji.replace('+', '')
            razem += float(n_istalacji)
            dl_zbioru_kategorii += 1
    sr_n_instalacji = razem / dl_zbioru_kategorii
    print(kategoria, ':', sr_n_instalacji)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Kategoria Communication posiada największą srednią liczbe instalacji, jest ona zawyzona przez takie giganty jak Waze czy Google Maps.

## Wnioski
Ten projekt zobrazował jak mniej więcej może wygladać workflow w pracy związanej z Data Science.